# Since nvcc uses gcc by default, appropriate flags must be passed to it
# instead of the ones generated by EB for icpc via CFLAGS.
# Trying to use icpc with nvcc ends in a __float128 undefined error.
# Author: Davide Vanzo (Vanderbilt University)
diff -ru magma-2.3.0.orig/Makefile magma-2.3.0/Makefile
--- magma-2.3.0.orig/Makefile	2018-11-26 16:09:17.782851847 -0600
+++ magma-2.3.0/Makefile	2018-11-26 17:31:39.526846548 -0600
@@ -10,7 +10,7 @@
 CC         ?= cc
 CXX        ?= c++
 NVCC       ?= nvcc
-FORT       ?=
+FORT       ?= $(FC)
 ifeq ($(FORT),)
     $(warning No Fortran compiler was given in FORT in make.inc. Some testers will not be able to check their results.)
 endif
@@ -23,17 +23,17 @@
 #FPIC       = -fPIC
 
 # may want -std=c99 for CFLAGS, -std=c++11 for CXXFLAGS
-CFLAGS     ?= -O3 $(FPIC) -DADD_ -Wall -MMD
-CXXFLAGS   ?= $(CFLAGS)
-NVCCFLAGS  ?= -O3         -DADD_ -Xcompiler "$(FPIC) -Wall -Wno-unused-function"
-FFLAGS     ?= -O3 $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument
-F90FLAGS   ?= -O3 $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument
-LDFLAGS    ?= -O3 $(FPIC)
+CFLAGS     := $(FPIC) -DADD_ -Wall -MMD $(CFLAGS)
+CXXFLAGS   := $(CFLAGS)
+NVCCFLAGS  :=         -DADD_ -Xcompiler " -Wall -Wno-unused-function -DADD_ -MMD -O2 -ftree-vectorize -march=native -fno-math-errno -fopenmp -fPIC"
+FFLAGS     := $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument $(FCFLAGS)
+F90FLAGS   := $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument $(F90FLAGS)
+LDFLAGS    := $(FPIC) -fopenmp $(LDFLAGS)
 
 INC        ?= -I$(CUDADIR)/include
 
 LIBDIR     ?= -L$(CUDADIR)/lib
-LIB        ?= -lcudart -lcublas -lcusparse -llapack -lblas
+LIB        := $(LIBLAPACK) -lcudart -lcublas -lcusparse $(LIBS)
 
 GPU_TARGET ?= Kepler Maxwell Pascal
 
Only in magma-2.3.0: Makefile~
